common/src/main/java/jp/co/soramitsu/common/compose/component/Toolbar.kt

```diff
@@ -0,0 +1,178 @@
+import androidx.compose.material.IconButton
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Alignment.Companion.End
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import jp.co.soramitsu.common.R
+import jp.co.soramitsu.common.compose.theme.backgroundBlurColor
+import jp.co.soramitsu.common.compose.theme.white
+import jp.co.soramitsu.common.compose.theme.customTypography
+import jp.co.soramitsu.common.compose.theme.FearlessTheme
+import jp.co.soramitsu.common.compose.theme.colorAccent
+
+
+data class MainToolbarViewState(
+    val title: String,
+    val selectedChainName: String,
+    val selectedChainId: String,
+    val selectedChainStatusColor: Color
+)
+
+data class MenuIconItem(
+    @DrawableRes val icon: Int,
+    val onClick: () -> Unit
+)
+
+@Composable
+fun MainToolbar(
+    state: MainToolbarViewState,
+    onChangeChainClick: () -> Unit,
+    @DrawableRes navigationIcon: Int? = null,
+    onNavigationClick: (() -> Unit)? = null,
+    menuItems: List<MenuIconItem>? = null
+) {
+    Row(
+        modifier = Modifier
+            .height(60.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp),
+        verticalAlignment = CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        Box(
+            contentAlignment = Alignment.CenterStart,
+            modifier = Modifier.weight(1f)
+        ) {
+            navigationIcon?.let {
+                IconButton(
+                    onClick = onNavigationClick ?: {},
+                    modifier = Modifier
+                        .clip(CircleShape)
+                        .background(backgroundBlurColor)
+                        .size(40.dp),
+                ) {
+                    Icon(
+                        painter = painterResource(id = navigationIcon),
+                        tint = white,
+                        contentDescription = null
+                    )
+                }
+            }
+        }
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .weight(2f),
+            horizontalAlignment = CenterHorizontally
+        ) {
+            Text(
+                text = state.title,
+                style = MaterialTheme.customTypography.header4,
+                maxLines = 1
+            )
+
+            MarginVertical(margin = 4.dp)
+
+            Row(
```

Comment:
Extract it to the separated ChainSelector so we can use it somewhere else if needed, it will also improve the readability of the composable function.

Reply:
Fixed